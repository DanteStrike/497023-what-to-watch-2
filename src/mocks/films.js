export const films = [
  {
    id: 1,
    name: `filmOne`,
    posterImage: `img/filmOne-poster.jpg`,
    preview: {
      image: `img/filmOne.jpg`,
      videoSrc: `https://some-linkOne`
    },
    background: {
      image: `img/filmOne.jpg`,
      color: `#ffffff`
    },
    videoSrc: `https://some-linkOne`,
    description: `Some text filmOne`,
    rating: 1.0,
    scoresCount: 1,
    director: `directorOne`,
    starring: [`actorOne`, `actorTwo`],
    runTime: 1,
    genre: `genreOne`,
    released: 2019,
    isFavorite: false,
  }, {
    id: 3,
    name: `filmTwo`,
    posterImage: `img/filmTwo-poster.jpg`,
    preview: {
      image: `img/filmTwo.jpg`,
      videoSrc: `https://some-linkTwo`
    },
    background: {
      image: `img/filmTwo.jpg`,
      color: `#000000`
    },
    videoSrc: `https://some-linkTwo`,
    description: `Some text filmTwo`,
    rating: 2.0,
    scoresCount: 2,
    director: `directorTwo`,
    starring: [`actorThree`, `actorFour`],
    runTime: 2,
    genre: `genreTwo`,
    released: 2020,
    isFavorite: true,
  }
];
