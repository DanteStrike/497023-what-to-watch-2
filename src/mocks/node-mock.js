export const createNodeMock = (element) => {
  if (element.type === `video`) {
    return {
      src: null,
      isMuted: null,
      poster: null,
      addEventListener: () => {}
    };
  }
  return null;
};
