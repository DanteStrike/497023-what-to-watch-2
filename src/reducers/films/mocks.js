const filmsRAW = [
  {
    "id": 1,
    "name": `filmOne`,
    "poster_image": `img/filmOne-poster.jpg`,
    "preview_image": `img/filmOne.jpg`,
    "background_image": `img/filmOne.jpg`,
    "background_color": `#ffffff`,
    "video_link": `https://some-linkOne`,
    "preview_video_link": `https://some-linkOne`,
    "description": `Some text filmOne`,
    "rating": 1.0,
    "scores_count": 1,
    "director": `directorOne`,
    "starring": [`actorOne`, `actorTwo`],
    "run_time": 1,
    "genre": `genreOne`,
    "released": 2019,
    "is_favorite": false,
  },
  {
    "id": 3,
    "name": `filmTwo`,
    "poster_image": `img/filmTwo-poster.jpg`,
    "preview_image": `img/filmTwo.jpg`,
    "background_image": `img/filmTwo.jpg`,
    "background_color": `#000000`,
    "video_link": `https://some-linkTwo`,
    "preview_video_link": `https://some-linkTwo`,
    "description": `Some text filmTwo`,
    "rating": 2.0,
    "scores_count": 2,
    "director": `directorTwo`,
    "starring": [`actorThree`, `actorFour`],
    "run_time": 2,
    "genre": `genreTwo`,
    "released": 2020,
    "is_favorite": true,
  }
];

const adaptedFilmsData = [
  {
    id: 1,
    name: `filmOne`,
    posterImage: `img/filmOne-poster.jpg`,
    preview: {
      image: `img/filmOne.jpg`,
      videoSrc: `https://some-linkOne`
    },
    background: {
      image: `img/filmOne.jpg`,
      color: `#ffffff`
    },
    videoSrc: `https://some-linkOne`,
    description: `Some text filmOne`,
    rating: 1.0,
    scoresCount: 1,
    director: `directorOne`,
    starring: [`actorOne`, `actorTwo`],
    runTime: 1,
    genre: `genreOne`,
    released: 2019,
    isFavorite: false,
  }, {
    id: 3,
    name: `filmTwo`,
    posterImage: `img/filmTwo-poster.jpg`,
    preview: {
      image: `img/filmTwo.jpg`,
      videoSrc: `https://some-linkTwo`
    },
    background: {
      image: `img/filmTwo.jpg`,
      color: `#000000`
    },
    videoSrc: `https://some-linkTwo`,
    description: `Some text filmTwo`,
    rating: 2.0,
    scoresCount: 2,
    director: `directorTwo`,
    starring: [`actorThree`, `actorFour`],
    runTime: 2,
    genre: `genreTwo`,
    released: 2020,
    isFavorite: true,
  }
];

const normalizedFilmsData = {
  byIDs: {
    "1": adaptedFilmsData[0],
    "3": adaptedFilmsData[1]
  },
  allIDs: [1, 3]
};

const store = {
  films: {
    data: normalizedFilmsData
  }
};

export default {
  filmsRAW,
  adaptedFilmsData,
  normalizedFilmsData,
  store
};
