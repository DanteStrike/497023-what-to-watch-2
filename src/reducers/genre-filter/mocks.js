import utils from "./utils";

const genreFilterStore = {
  currentFilter: utils.ALL_GENRE,
  data: {
    genres: [`All genre`, `genreOne`, `genreTwo`],
    byGenres: {
      "All genre": [1, 2, 3, 4],
      "genreOne": [2, 4],
      "genreTwo": [1, 3]
    }
  }
};

const store = {
  genreFilter: genreFilterStore
};

export default {
  genreFilterStore,
  store
};
