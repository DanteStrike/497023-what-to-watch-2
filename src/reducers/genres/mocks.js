import utils from "./utils";
import StoreNameSpace from "../store-name-space";

const genreFilterStore = {
  currentFilter: utils.ALL_GENRE,
  data: {
    genres: [`All genre`, `genreOne`, `genreTwo`],
    byGenres: {
      "All genre": [1, 2, 3, 4],
      "genreOne": [2, 4],
      "genreTwo": [1, 3]
    }
  }
};

const store = {
  [StoreNameSpace.GENRES]: genreFilterStore
};

export default {
  genreFilterStore,
  store
};
