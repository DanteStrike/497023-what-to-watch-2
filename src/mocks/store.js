import StoreNameSpace from "../reducers/store-name-space.js";
import {comments} from "./comments.js";
import {normalizedFilmsData} from "./films";

export const initStore = {
  [StoreNameSpace.APP]: {
    isReady: false,
    videoPlayerFilmID: -1
  },

  [StoreNameSpace.FILMS]: {
    data: {
      allIDs: [],
      byIDs: {}
    },

    promo: {
      filmID: null
    }
  },

  [StoreNameSpace.COMMENTS]: {
    postCommentState: {
      isSuccess: false,
      error: {
        isError: false,
        msg: ``
      }
    },
    curFilmComments: []
  },

  [StoreNameSpace.GENRES]: {
    currentFilter: `All genre`,
    data: {
      genres: [`All genre`],
      byGenres: {
        "All genre": [],
      }
    }
  },

  [StoreNameSpace.CATALOG]: {
    displayedFilmsAmount: 0
  },

  [StoreNameSpace.USER]: {
    auth: {
      isAuth: false,
      error: {
        isError: false,
        target: ``,
        msg: ``
      }
    },

    toggleFavoriteStatus: {
      isSuccess: false,
      error: {
        isError: false
      }
    },

    data: {
      id: null,
      email: ``,
      name: ``,
      avatarUrl: ``,
      myListFilmsIDs: []
    }
  }
};

export const loadedStore = {
  [StoreNameSpace.APP]: {
    isReady: true,
    videoPlayerFilmID: 1
  },

  [StoreNameSpace.FILMS]: {
    data: normalizedFilmsData,

    promo: {
      filmID: 3
    }
  },

  [StoreNameSpace.COMMENTS]: {
    postCommentState: {
      isSuccess: true,
      error: {
        isError: false,
        msg: ``
      }
    },
    curFilmComments: comments
  },

  [StoreNameSpace.GENRES]: {
    currentFilter: `All genre`,
    data: {
      genres: [`All genre`, `genreOne`, `genreTwo`],
      byGenres: {
        "All genre": [1, 3],
        "genreOne": [1],
        "genreTwo": [3]
      }
    }
  },

  [StoreNameSpace.CATALOG]: {
    displayedFilmsAmount: 0
  },

  [StoreNameSpace.USER]: {
    auth: {
      isAuth: true,
      error: {
        isError: true,
        target: `target`,
        msg: `any`
      }
    },

    toggleFavoriteStatus: {
      isSuccess: true,
      error: {
        isError: false
      }
    },

    data: {
      id: 1,
      email: `Oliver.conner@gmail.com`,
      name: `Oliver.conner`,
      avatarUrl: `img/1.png`,
      myListFilmsIDs: [3]
    }
  },
};
