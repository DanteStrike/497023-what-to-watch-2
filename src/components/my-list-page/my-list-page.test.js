import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from "react-router-dom";
import MyListPage from "./my-list-page.jsx";

import {Provider} from "react-redux";
import {createStore} from "redux";

import {loadedStore} from "../../mocks/store.js";
import {createNodeMock} from "../../mocks/node-mock";

it(`Render correctly MyListPage component`, () => {
  const store = createStore(() => loadedStore);
  const component = renderer
    .create(
        <Provider store={store}>
          <Router>
            <MyListPage/>
          </Router>
        </Provider>,
        {
          createNodeMock
        }
    ).toJSON();

  expect(component).toMatchSnapshot();
});
