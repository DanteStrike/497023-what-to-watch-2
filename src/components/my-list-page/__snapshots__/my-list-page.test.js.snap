// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly MyListPage component 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src="img/1.png"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "willChange": "transform",
          }
        }
      >
        <a
          href="/films/3"
          onClick={[Function]}
          style={
            Object {
              "color": "unset",
              "textDecoration": "none",
            }
          }
        >
          <div
            className="small-movie-card__image"
          >
            <video
              style={
                Object {
                  "backgroundColor": "black",
                  "display": "block",
                  "height": "100%",
                  "left": "50%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "position": "absolute",
                  "top": "50%",
                  "transform": "translate(-50%, -50%)",
                  "width": "100%",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            filmTwo
          </h3>
        </a>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
